src/main/java/com/marklogic/spark/reader/optic/OpticReadContext.java

```diff
@@ -130,20 +172,55 @@ private PlanBuilder.Plan buildPlanForBucket(RowManager rowManager, PlanAnalysis.
             .bindParam("ML_LOWER_BOUND", bucket.lowerBound)
             .bindParam("ML_UPPER_BOUND", bucket.upperBound);
 
+        // Add user-defined parameters from options
+        for(Map.Entry<String, String> entry : this.userDefinedParams.entrySet()) {
```

Comment:
It looks like `userDefinedParams` is not needed - yet - since the tests do not make use of this - i.e. I can remove all of this and the tests will still pass.

Reply:
I will honestly have to look through my code and figure out what it's used for.  I'll let you know if I figure it out :)